File: tests/test_r/models/order.R
source("models/user.R")

create_order <- function(id, user_id, amount) {
    order <- list(
        id = id,
        user_id = user_id,
        amount = amount
    )
    class(order) <- "Order"
    return(order)
}

order_total <- function(order) {
    return(order$amount)
}

File: tests/test_haskell/Models/Order.hs
module Models.Order
    ( Order(..)
    , mkOrder
    , orderTotal
    ) where

import Models.User

data Order = Order
    { orderId     :: Int
    , orderUserId :: Int
    , orderAmount :: Double
    } deriving (Show, Eq)

mkOrder :: Int -> Int -> Double -> Order
mkOrder oid uid amount = Order
    { orderId = oid
    , orderUserId = uid
    , orderAmount = amount
    }

orderTotal :: Order -> Double
orderTotal = orderAmount
